project (sprint4)

Irina2107 opened the pull request:
Hi.
Please review.

Commit:
add git ignore

Commit:
add pom.xml

Commit:
add ApConf and BaseTest

Commit:
add pages

Commit:
add testes

alesyaisachenko commented on src/main/java/ru/praktikum/scooter/test/config/AppConfig.java:
```diff
@@ -0,0 +1,5 @@
+package ru.praktikum.scooter.test.config;
+
+public class AppConfig {
+    public final static String AppUrl = "https://qa-scooter.praktikum-services.ru";}
```
Это нужно доработать:

AppUrl -- нужно APP_URL, константы именуются в таком формате

alesyaisachenko commented on src/main/java/ru/praktikum/scooter/test/pageobject/PageCreatedOrder.java:
```diff
@@ -0,0 +1,43 @@
+package ru.praktikum.scooter.test.pageobject;
```
Это нужно исправить:
pageobject -- нужно page_object, пакеты именуются таким форматом

alesyaisachenko commented on src/test/java/ru/praktikum/scooter/test/CheckAccordionTest.java:
```diff
@@ -0,0 +1,89 @@
+            pageCreatedOrder.clickAgreeToCook();
+
+            ////Ожидание прогрузки страницы
+            By forWait = By.xpath(".//div[contains(text(),'Вопросы о важном')]");
```
Это нужно доработать:

в тестах не должно быть локаторов поиска.

Локаторы поиска и методы работы с ними должны быть на соответствующей Page.

Подправь, плиз, этот момент и в других тестовых классах.

alesyaisachenko commented on src/test/java/ru/praktikum/scooter/test/CheckAccordionTest.java:
```diff
@@ -0,0 +1,89 @@
+            pageCreatedOrder = new PageCreatedOrder(driver);
+
+            String expectedButton = "Сколько это стоит? И как оплатить?";
+            String expectedTextOfAccordion = "Сутки — 400 рублей. Оплата курьеру — наличными или картой.";
```
Это нужно доработать:

текст вопроса и ответа нужно хранить как константу (static final).

Ответ можно хранить в MainPage или в тесте.

alesyaisachenko commented on src/test/java/ru/praktikum/scooter/test/CheckAccordionTest.java:
```diff
@@ -0,0 +1,89 @@
+
+
+        @Test
+        public void AccordionCommon1() {
```
Можно лучше:

эти тесты можно было реализовать через параметризацию

alesyaisachenko commented on src/test/java/ru/praktikum/scooter/test/CheckAccordionTest.java:
```diff
@@ -0,0 +1,89 @@
+            assertEquals("Текст не совпадает", expectedTextOfAccordion, actualTextOfAccordion);
+        }
+        @Test
+        public void AccordionCommon2() {
```
Это нужно доработать:

1. AccordionCommon2() -- методы именуются таким форматом accordionCommon2().

Подправь, плиз, нейминг всех тестов.

2. AccordionCommon2() -- не удачный нейминг, из названия теста должно быть ясно, что тестируется внутри и чем один тест отличается от другого. Плюс не принято использовать числа при нейминге, код с числами тяжело поддерживать другим программистам.

Нужно переименовать все тесты такого формата.

alesyaisachenko commented on src/test/java/ru/praktikum/scooter/test/BaseTestScooterOrder.java:
```diff
@@ -0,0 +1,27 @@
+public class BaseTestScooterOrder {
+    WebDriver driver;
+    @Before
+    public void setUp() {
```
Хорошая идея с BaseTest!

alesyaisachenko commented on src/test/java/ru/praktikum/scooter/test/TestScooterOrderYes.java:
```diff
@@ -0,0 +1,65 @@
+
+        assertTrue("Нет окна. Ошибка", ModalOrderDisplayed.isDisplayed());
+    }
+
```
Это нужно доработать:

Нужно написать еще один тест, который будет кликать по другой кнопке "Заказать"

alesyaisachenko commented on src/main/java/ru/praktikum/scooter/test/pageobject/PageCreatedOrder.java:
```diff
@@ -0,0 +1,43 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+
+public class PageCreatedOrder {
```
Можно лучше:

эту страницу лучше назвать MainPage

Commit:
Hi. Please, check. I tried to fix all comments.
Added new classes PageStatusOrder and testAccordionTextAnswerMapping. Deleted - CheckAccordionTest.

alesyaisachenko commented on src/test/java/ru/praktikum/scooter/test/baseTestScooterOrder.java:
```diff
@@ -0,0 +1,27 @@
+import java.io.File;
+import static ru.praktikum.scooter.test.config.AppConfig.APP_URL;
+
+public class baseTestScooterOrder {
```
Это нужно обязательно исправить:




baseTestScooterOrder  -- нужно BaseTestScooterOrder  , классы именуются в таком формате.




Подправь, плиз, этот момент во всем коде.

alesyaisachenko commented on src/test/java/ru/praktikum/scooter/test/testAccordionTextAnswerMapping.java:
```diff
@@ -0,0 +1,85 @@
+      static final String expectedSecondAnswerText = "Пока что у нас так: один заказ — один самокат. Если хотите покататься с друзьями, можете просто сделать несколько заказов — один за другим.";
+
+      @Test
+      public void testAnswerByFirstButton() {
```
Это нужно обязательно исправить:

нужно написать тесты на все вопросы, а не только на два.

alesyaisachenko commented on src/test/java/ru/praktikum/scooter/test/testAccordionTextAnswerMapping.java:
```diff
@@ -0,0 +1,85 @@
+
+public class testAccordionTextAnswerMapping extends baseTestScooterOrder {
+      MainPage mainPage;
+      static final String expectedFirstQuestionText = "Сколько это стоит? И как оплатить?";
```
Это нужно обязательно исправить:

static final String expectedFirstQuestionText -- нужно static final String EXPECTED_FIRST_QUESTION_TEXT (константы именуются в таком формате (static final )).

Подправь, плиз, этот момент и в аналогичных случаях

Commit:
Hi. Please, check after fixed comment.

alesyaisachenko commented on src/test/java/ru/praktikum/scooter/test/TestScooterOrderNoModalOrderNotDisplayed.java:
```diff
@@ -0,0 +1,90 @@
+    String PhoneNumber = "+79650084939";
+    String OrderCommentText = "После 22:00";
+    @Test
+        public void OrderButtonUpOrderNotCreated()  {
```
Это нужно обязательно доработать:

OrderButtonUpOrderNotCreated() -- нужно orderButtonUpOrderNotCreated(), методы именуются в таком формате.
